### PR TITLE
WIP (wait for sqlalchemy2-stubs): Added types to lib/galaxy/model/__init__.py

### PR DESCRIPTION
**Pull request by dakariakin**
Why: common-workflow-lab#137
What: Improved type hints

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/model/__init__.py [hunk 1]
```diff
@@ -321,6 +329,8 @@ def get_display_name(self):
 
 
 class UsesCreateAndUpdateTime:
+    update_time: datetime
+    create_time: datetime
```
Comment (mvdbeek) [inline lib/galaxy/model/__init__.py:333]: That'll conflict with the types defined by the sqlalchemy models

File: lib/galaxy/model/__init__.py [hunk 2]
```diff
@@ -309,6 +316,7 @@ def _serialize(self, id_encoder: IdEncodingHelper, serialization_options: Serial
 
 
 class HasName:
+    name: str
```
Comment (mvdbeek) [inline lib/galaxy/model/__init__.py:319]: That'll conflict with the types defined by the sqlalchemy models

File: lib/galaxy/model/__init__.py [hunk 3]
```diff
@@ -374,6 +384,10 @@ def cached_id(galaxy_model_object):
 
 class JobLike:
     MAX_NUMERIC = 10**(JOB_METRIC_PRECISION - JOB_METRIC_SCALE) - 1
+    _numeric_metric: Callable
+    _text_metric: Callable
+    id: int
+    tool_id: int
```
Comment (mvdbeek) [inline lib/galaxy/model/__init__.py:390]: That'll conflict with the types defined by the sqlalchemy models

**Review by mvdbeek — COMMENTED** [review]
I would advise against typing `model/__init__.py` for now (or at least everything that is backed by a database column), I think this will heavily conflict with typing when the sqlalchemy2-stubs are installed (which don't seem to be quite ready yet).

**Comment by dakariakin** [comment]
@mvdbeek Thank you, that makes sense. I'll update this PR after sqlalchemy2-stubs installed

**Comment by dakariakin** [comment]
The original issue is closed https://github.com/common-workflow-lab/galaxy/issues/137

**Pull request closed without merging**